rotkehlchen/data_import/importers/kucoin.py

```diff
@@ -0,0 +1,113 @@
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Location, TradeType
+
+
+class KucoinFileType(Enum):
+    """The type of the file we are parsing"""
+    TRADES = auto()  # default header format
+    TRADES_OLD = auto()  # old header format
+
+
+class KucoinImporter(BaseExchangeImporter):
+    @staticmethod
+    def _consume_trades(
+            csv_row: dict[str, str],
+            file_type: KucoinFileType,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> Trade:
+        """
+        Consume entries of kucoin order history csv
+
+        May raise:
+        - KeyError
+        - DeserializationError
+        """
+        if file_type == KucoinFileType.TRADES:
+            date = csv_row['tradeCreatedAt']
+            base, quote = csv_row['symbol'].split('-')
+            trade_type = TradeType.BUY if csv_row['side'] == 'buy' else TradeType.SELL
+            amount = csv_row['size']
+            rate = csv_row['price']
+            fee = get_key_if_has_val(csv_row, 'fee')
+            fee_currency = get_key_if_has_val(csv_row, 'feeCurrency')
+        else:
+            date = csv_row['Time']
+            base, quote = csv_row['Pair'].split('/')
+            trade_type = TradeType.BUY if csv_row['Side'] == 'Buy' else TradeType.SELL
+            amount = csv_row['Amount']
+            rate = csv_row['FilledPrice']
+            fee = get_key_if_has_val(csv_row, 'Fee')
+            fee_currency = get_key_if_has_val(csv_row, 'FeeCoin')
+
+        return Trade(
+            timestamp=deserialize_timestamp_from_date(
+                date=date,
+                formatstr=timestamp_format,
+                location='Kucoin order history import',
+            ),
+            location=Location.KUCOIN,
+            base_asset=asset_from_kucoin(base),
+            quote_asset=asset_from_kucoin(quote),
+            trade_type=trade_type,
+            amount=deserialize_asset_amount(amount),
+            rate=deserialize_price(rate),
+            fee=deserialize_fee(fee),
+            fee_currency=asset_from_kucoin(fee_currency) if fee_currency else None,
+        )
+
+    def _import_csv(self, write_cursor: DBCursor, filepath: Path, **kwargs: Any) -> None:
+        """
+        Import trades from Kucoin. Find out which file
+        we are parsing depending on its format.
+
+        May raise:
+        - UnsupportedCSVEntry if operation not supported
+        - InputError if a column we need is missing
+        """
+        with open(filepath, encoding='utf-8-sig') as csvfile:
+            while True:
+                saved_pos = csvfile.tell()
+                line = csvfile.readline()
+                if not (line.startswith(('Name: ', 'UserID: ', 'Date Generated: ')) or line.strip() == ''):  # noqa: E501
```

Comment:
why this line here? could you add a comment here explaining why this is needed?

Reply:
I see it's not needed, so removing it